### ISSUE TITLE
duplicate generation of labels when variables are used.

### ISSUE THREAD

**Issue by ymark** [isl/x3ml#122]
If we use variables for linking resources found in different links, then they are properly linked (i.e. only one URI is created and re-used), however if a label generator is defined as well, then multiple labels will be generated.

Using the following resources as input: 

```
<dataroot>
    <COIN>
        <ID>100</ID>
		<CODE>CODE-100</CODE>
        <COUNTRY_ID>C1</COUNTRY_ID>
    </COIN>
</dataroot>
```

and the following X3ML mapping resources: 

```
<x3ml version="1.0" source_type="xpath">
    <namespaces>
        <namespace prefix="crm" uri="http://www.cidoc-crm.org/cidoc-crm/"/>
        <namespace prefix="rdf" uri="http://www.w3.org/1999/02/22-rdf-syntax-ns#"/>
        <namespace prefix="rdfs" uri="http://www.w3.org/2000/01/rdf-schema#"/>
    </namespaces>
    <mappings>
        <mapping>
            <domain>
                <source_node>//COIN</source_node>
                <target_node>
                    <entity>
                        <type>crm:E22_Man-Made_Object</type>
                        <instance_generator name="UUID"/>
                    </entity>
                </target_node>
            </domain>
            <link>
                <path>
                    <source_relation><relation>ID</relation></source_relation>
                    <target_relation>
                        <relationship>crm:P1_is_identified_by</relationship>
                    </target_relation>
                </path>
                <range>
                    <source_node>ID</source_node>
                    <target_node>
                        <entity variable="v1">
                            <type>crm:E41_Appellation</type>
                            <instance_generator name="UUID"/>
                            <label_generator name="Literal">
                                <arg name="text">text()</arg>
                            </label_generator>
                        </entity>
                    </target_node>
                </range>
            </link>
			<link>
                <path>
                    <source_relation><relation>CODE</relation></source_relation>
                    <target_relation>
                        <relationship>crm:PX_is_also_identified_by</relationship>
                    </target_relation>
                </path>
                <range>
                    <source_node>CODE</source_node>
                    <target_node>
                        <entity variable="v1">
                            <type>crm:E41_Appellation</type>
                        </entity>
                    </target_node>
                </range>
            </link>
        </mapping>
    </mappings>
</x3ml>
```

Will generate the following RDF output: 

```
<rdf:RDF
    xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
    xmlns:crm="http://www.cidoc-crm.org/cidoc-crm/"
    xmlns:skos="http://www.w3.org/2004/02/skos/core#"
    xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#">
  <crm:E22_Man-Made_Object rdf:about="uuid:AA">
    <crm:PX_is_also_identified_by>
      <crm:E41_Appellation rdf:about="uuid:AB">
        <rdfs:label>CODE-100</rdfs:label>
        <rdfs:label>100</rdfs:label>
      </crm:E41_Appellation>
    </crm:PX_is_also_identified_by>
    <crm:P1_is_identified_by rdf:resource="uuid:AB"/>
  </crm:E22_Man-Made_Object>
</rdf:RDF>
```

Notice that the UUID `uuid:AB` has been properly re-used however two labels are generated (w.r.t the value of the source_node).

The above resources for reproducing the issue are attached.
[mappings.x3ml.txt](https://github.com/isl/x3ml/files/1946480/mappings.x3ml.txt)
[input.xml.txt](https://github.com/isl/x3ml/files/1946481/input.xml.txt)

**Comment by ymark** [comment]
The reason for this is because technically the generation of labels is out of the scope of variables. Therefore when a variable is used then the instance will not be generated again (as it should), however the label will be generated using the input sources.

Independently of the discussion if it is an issue or not, it might confuse users. So one solution to overcome this is to prevent using label-generators for entities that are referenced using a variable (and clearly their generator details are not explicitly given).

**Comment by ymark** [comment]
I commented the sources responsible for copying the label_generators for entities that use referenced (using a variable) generators. The corresponding commit showing them is 69826da5e5a8615c8ea04731b1bdbfa4f1b43529

**Comment by ymark** [comment]
Now the result will contain only the label generator in the place where the label is explicitly defined and re-used everywhere

```
<rdf:RDF
    xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
    xmlns:crm="http://www.cidoc-crm.org/cidoc-crm/"
    xmlns:skos="http://www.w3.org/2004/02/skos/core#"
    xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#">
  <crm:E22_Man-Made_Object rdf:about="uuid:AA">
    <crm:PX_is_also_identified_by>
      <crm:E41_Appellation rdf:about="uuid:AB">
        <rdfs:label>100</rdfs:label>
      </crm:E41_Appellation>
    </crm:PX_is_also_identified_by>
    <crm:P1_is_identified_by rdf:resource="uuid:AB"/>
  </crm:E22_Man-Made_Object>
</rdf:RDF>
```

The updated will be included in X3ML engine version 1.8.5
